examples/darwin-framework-tool/commands/clusters/ReportCommandBridge.h

```diff
@@ -237,7 +277,81 @@ class SubscribeEvent : public ModelCommand {
     chip::Optional<bool> mKeepSubscriptions;
     chip::Optional<bool> mAutoResubscribe;
     chip::Optional<chip::EventNumber> mEventNumber;
+    chip::Optional<bool> mIsUrgent;
     bool mSubscriptionEstablished = NO;
     uint16_t mMinInterval;
     uint16_t mMaxInterval;
+
+    void Shutdown() override
+    {
+        mSubscriptionEstablished = NO;
+        ModelCommand::Shutdown();
+    }
+
+    bool DeferInteractiveCleanup() override { return mSubscriptionEstablished; }
+
+private:
+    chip::ClusterId mClusterId = chip::kInvalidClusterId;
+    chip::EventId mEventId;
+};
+
+class ReadEvent : public ModelCommand {
+public:
+    ReadEvent()
+        : ModelCommand("read-event-by-id")
+    {
+        AddArgument("cluster-id", 0, UINT32_MAX, &mClusterId);
+        AddArgument("event-id", 0, UINT32_MAX, &mEventId);
+        AddArgument("event-min", 0, UINT64_MAX, &mEventNumber);
+        ModelCommand::AddArguments();
+    }
+
+    ReadEvent(chip::ClusterId clusterId)
+        : ModelCommand("read-event-by-id")
+        , mClusterId(clusterId)
+    {
+        AddArgument("event-id", 0, UINT32_MAX, &mEventId);
+        AddArgument("event-min", 0, UINT64_MAX, &mEventNumber);
+        ModelCommand::AddArguments();
+    }
+
+    ~ReadEvent() {}
+
+    CHIP_ERROR SendCommand(MTRBaseDevice * _Nonnull device, chip::EndpointId endpointId) override
+    {
+        dispatch_queue_t callbackQueue = dispatch_queue_create("com.chip.command", DISPATCH_QUEUE_SERIAL);
+        MTRReadParams * params = [[MTRReadParams alloc] init];
+        if (mFabricFiltered.HasValue()) {
+            params.filterByFabric = mFabricFiltered.Value();
+        }
+        if (mEventNumber.HasValue()) {
+            params.eventMin = [NSNumber numberWithUnsignedLongLong:mEventNumber.Value()];
+        }
+
+        [device readEventsWithEndpointID:[NSNumber numberWithUnsignedShort:endpointId]
+                               clusterID:[NSNumber numberWithUnsignedInteger:mClusterId]
+                                 eventID:[NSNumber numberWithUnsignedInteger:mEventId]
```

Comment:
Again, the wildcard handling here is off.  Same followup as above.

Reply:
@bzbarsky-apple This is current status. If more changes are needed, please let us know:
1. Arguments of 0xFFFF is working as wildcards like chip-tool
2. To support more wildcard logics same to chip-tool, "any" cluster is added